src/components/my-reading/saved-bookmarks.js

```diff
@@ -0,0 +1,312 @@
+      published_date: Number(`${publishedDate}000`),
+      releaseBranch,
+      slug,
+      is_bookmarked: true,
+      toggle_bookmark: handleToggleBookmark,
+    }
+  }
+
+  const getBookmarks = async page => {
+    const { payload } = await getMultipleBookmarks(
+      jwt,
+      userID,
+      (page - 1) * BOOKMARK_PER_PAGE,
+      BOOKMARK_PER_PAGE
+    )
+    setTotalSavedBookmarkCount(_.get(payload, 'data.meta.total', 0))
+    const { records } = _.get(payload, 'data', [])
+    if (records.length === 0) {
+      setShowEmptyState(true)
+    } else {
+      setShowEmptyState(false)
+      setBookmarks(_.map(records, bookmark => filterBookmark(bookmark)))
+    }
+  }
+
+  useEffect(() => {
+    if (bookmarks.length === 0 && page > 1) {
+      const redirectPage = Math.max(page === totalPages ? page - 1 : page, 1)
+      navigate.push(
+        `${routes.myReadingPage.savedBookmarksPage.path}?page=${redirectPage}`
+      )
+    }
+  }, [bookmarks])
+
+  useEffect(() => {
+    if (totalSavedBookmarkCount === 0) {
+      setShowEmptyState(true)
+    }
+  }, [totalSavedBookmarkCount])
+
+  useEffect(() => {
+    getBookmarks(page)
+  }, [page])
+
+  return (
+    <Container>
+      <Title1
+        title={'已收藏'}
+        subtitle={
+          totalSavedBookmarkCount ? `共${totalSavedBookmarkCount}篇` : null
+        }
+      />
+      {showEmptyState ? (
+        <EmptyStateConatiner>
+          <EmptyState
+            style={EmptyState.Style.DEFAULT}
+            title="你還沒有收藏任何報導"
+            guide={
+              <>
+                <P2 text="點擊" />
+                <Bookmark
+                  type={Bookmark.Type.ADD}
+                  releaseBranch={releaseBranch}
+                />
+                <P2 text="將喜愛的文章加入我的書籤" />
+              </>
+            }
+            buttonText="前往首頁"
+            buttonUrl={requestOrigin.forClientSideRendering[releaseBranch].main}
+            releaseBranch={releaseBranch}
+          />
+        </EmptyStateConatiner>
+      ) : (
+        <>
+          <ListContainer>
+            <CardList
+              isFetching={!isDeleting && isFetching}
+              showSpinner={!isDeleting && isFetching}
+              data={bookmarks}
+              showIsBookmarked={true}
+              width={100}
+            />
+          </ListContainer>
+          <PaginationContainer>
+            <Pagination currentPage={page} totalPages={totalPages} />
+          </PaginationContainer>
+          <SnackBarContainer showSnackBar={showSnackBar}>
+            <SnackBarDiv>
+              <SnackBar text={snackBarText} />
+            </SnackBarDiv>
+          </SnackBarContainer>
+        </>
+      )}
+    </Container>
+  )
+}
+
+SavedBookmarks.propTypes = {
+  isDeleting: PropTypes.bool,
+  isFetching: PropTypes.bool,
+  jwt: PropTypes.string,
+  userID: PropTypes.number,
+  getMultipleBookmarks: PropTypes.func,
+  page: PropTypes.number,
+  totalPages: PropTypes.number,
```

Comment:
Could we mark `isRequired` for props which should always have values

Reply:
fixed

Reply:
also `getMultipleBookmarks`, `isDeleting`, `isFetching` ?

Reply:
fixed